client/lib/accept/dialog.jsx

```diff
@@ -29,11 +29,9 @@ const AcceptDialog = React.createClass( {
 	},
 
 	onClose: function( action ) {
-		this.props.onClose( 'accept' === action );
-
-		if ( this.isMounted() ) {
-			this.setState( { isVisible: false } );
-		}
+		this.setState( { isVisible: false }, () => {
+			this.props.onClose( 'accept' === action );
+		} );
```

Comment:
do you know why `isMounted()` was necessary before?

Reply:
My best guess: the original author wanted to avoid calling `setState` on an unmounted component.

Reply:
I decided on dropping this change, as it's non-essential to this PR.

Reply:
The `isMounted` check would make sense if the `onClose` callback was called asynchronously:
```js
setTimeout( () => this.props.onClose(), 1000 );
```
Then we need to check if the component still exists at the time of the call. But in this case, everything is sync, so the call is not needed.

Instead of using the `setState` callback, this should work equally well:
```js
this.setState( { isVisible: false } );
this.props.onClose( ... );
```
Or even better -- it might avoid doing two rerenders instead of one.